entrypoint

```diff
@@ -114,10 +120,12 @@ class Entrypoint(object):
         # Presets, overriding or complementing defaults:
         preset = self.extenv.get("INPUT_PRESET")
         if preset:
-            if preset.startswith("clang-sanitize-"):
-                sanitizer = re.match("clang-sanitize-(.*)", preset).group(1)
-                self.cmd_cmake += " -DCMAKE_C_COMPILER=clang"
-                self.cmd_cmake += " -DCMAKE_CXX_COMPILER=clang++"
+            if preset.split("-")[:2] in [["clang", "sanitize"], ["gcc", "sanitize"]]:
+                match = re.match("(.*)-sanitize-(.*)", preset)
```

Comment:
I guess we could move this regex up, and match the compiler with something like `(gcc|clang)-sanitize-(.*)` and use the match in the "if"
`CompilerGroup` could also just be a dictionary